pyqtgraph/functions.py

```diff
@@ -2308,26 +2308,84 @@ def isosurface(data, level):
     return vertexes, faces
 
 
-    
+# Ad-hoc hard coded inverse of 3x3 matrix
+# taken from:
+# https://github.com/numpy/numpy/issues/17166#issue-686375258
+
+# Helper function for determinant
+def vdet(A):
+    detA = np.zeros_like(A[0, 0])
+    detA = A[0, 0] * (A[1, 1] * A[2, 2] - A[1, 2] * A[2, 1]) -\
+           A[0, 1] * (A[2, 2] * A[1, 0] - A[2, 0] * A[1, 2]) +\
+        A[0, 2] * (A[1, 0] * A[2, 1] - A[2, 0] * A[1, 1])
+    return detA
+
+
+# stolen verbatim from:
+def hdinv(A):
+    invA = np.zeros_like(A)
+    detA = vdet(A)
+
+    # naive singularity detection; be compat with ``QTransform.inverted()``.
+    if detA == 0:
+        return np.eye(3), False
+
+    invA[0, 0] = (-A[1, 2] * A[2, 1] +
+                  A[1, 1] * A[2, 2]) / detA
+    invA[1, 0] = (A[1, 2] * A[2, 0] -
+                  A[1, 0] * A[2, 2]) / detA
+    invA[2, 0] = (-A[1, 1] * A[2, 0] +
+                  A[1, 0] * A[2, 1]) / detA
+    invA[0, 1] = (A[0, 2] * A[2, 1] -
+                  A[0, 1] * A[2, 2]) / detA
+    invA[1, 1] = (-A[0, 2] * A[2, 0] +
+                  A[0, 0] * A[2, 2]) / detA
+    invA[2, 1] = (A[0, 1] * A[2, 0] -
+                  A[0, 0] * A[2, 1]) / detA
+    invA[0, 2] = (-A[0, 2] * A[1, 1] +
+                  A[0, 1] * A[1, 2]) / detA
+    invA[1, 2] = (A[0, 2] * A[1, 0] -
+                  A[0, 0] * A[1, 2]) / detA
+    invA[2, 2] = (-A[0, 1] * A[1, 0] +
+                  A[0, 0] * A[1, 1]) / detA
+
+    return invA, True
+
+
+from .debug import Profiler
+
+
 def invertQTransform(tr):
     """Return a QTransform that is the inverse of *tr*.
-    Rasises an exception if tr is not invertible.
-    
+    Raises an exception if tr is not invertible.
+
     Note that this function is preferred over QTransform.inverted() due to
```

Comment:
:facepalm:, I guess the reason is right here lol.

Clearly need to read docs summaries better :smile: